src/browser/base/content/ZenUIManager.mjs

```diff
@@ -593,4 +596,95 @@ var gZenVerticalTabsManager = {
     }
     target.appendChild(child);
   },
+
+  _insertDoubleClickListenerPinnedTabs() {
```

Comment:
This woudn't work if we add pinned tabs AFTER setting the listeners. I recommend adding `gBrowser.tabContainer.addEventListener()` and check if the originalTarget is the tab label

Reply:
No need. It adds this listener to every single tab. `renameTabStart` checks to make sure the tab is pinned.

Reply:
Done :+1: